src/gpio/gpio.c

```diff
@@ -73,6 +75,14 @@ mraa_gpio_init_internal(mraa_adv_func_t* func_table, int pin)
     dev->advance_func = func_table;
     dev->pin = pin;
 
+    // GPIO sysfs path may store the default path names schema or any other specific to vendor
+    dev->gpio_path = (char*) calloc(MAX_SIZE, sizeof(char));
+    if (dev->gpio_path == NULL) {
+        syslog(LOG_CRIT, "gpio%i: Failed to allocate memory for context's attribute", pin);
```

Comment:
What I mean is that instead of `"gpio%i: Failed to allocate memory for context's attribute"` make it a `"gpio%i: Failed to allocate memory for gpio_path attribute"` to make it clear, which exact piece were we not able to initialize.